powerstore/resource_snapshotrule_test.go

```diff
@@ -0,0 +1,448 @@
+		},
+		{
+			Config:      SnapshotRuleParamsWithInvalidNasAccessType,
+			ExpectError: regexp.MustCompile("Attribute nas_access_type must be one of these"),
+		},
+	}
+
+	for i := range tests {
+		resource.Test(t, resource.TestCase{
+			PreCheck:                 func() { testAccPreCheck(t) },
+			ProtoV6ProviderFactories: testProviderFactory,
+			Steps:                    []resource.TestStep{tests[i]},
+		})
+	}
+}
+
+// Test to Create SnapShotRule with empty string for some params
+func TestAccSnapshotRule_CreateSnapShotRuleWithEmptyString(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	tests := []resource.TestStep{
+		{
+			Config: SnapshotRuleParamsWithEmptyStringTimeZone,
+			Check: resource.ComposeTestCheckFunc(
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "name", "test_snapshotrule"),
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "time_of_day", "21:00"),
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "timezone", ""),
+				resource.TestCheckTypeSetElemAttr("powerstore_snapshotrule.test", "days_of_week.*", "Monday"),
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "desired_retention", "56"),
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "nas_access_type", "Snapshot"),
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "is_read_only", "false"),
+			),
+		},
+		{
+			Config: SnapshotRuleParamsWithEmptyStringNasAccessType,
+			Check: resource.ComposeTestCheckFunc(
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "name", "test_snapshotrule"),
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "time_of_day", "21:00"),
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "timezone", "UTC"),
+				resource.TestCheckTypeSetElemAttr("powerstore_snapshotrule.test", "days_of_week.*", "Monday"),
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "desired_retention", "56"),
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "nas_access_type", ""),
+				resource.TestCheckResourceAttr("powerstore_snapshotrule.test", "is_read_only", "false"),
+			),
+		},
+	}
+
+	for i := range tests {
+		resource.Test(t, resource.TestCase{
+			PreCheck:                 func() { testAccPreCheck(t) },
+			ProtoV6ProviderFactories: testProviderFactory,
+			Steps:                    []resource.TestStep{tests[i]},
+		})
+	}
+}
+
+// Test to Create SnapShotRule with mutually exclusive params available
+func TestAccSnapshotRule_CreateSnapShotRuleWithTimeOfdayandInterval(t *testing.T) {
+	if os.Getenv("TF_ACC") == "" {
+		t.Skip("Dont run with units tests because it will try to create the context")
+	}
+
+	resource.Test(t, resource.TestCase{
+		// PreCheck:                 func() { testAccPreCheck(t) },
```

Comment:
if this statement is not required, please remove it

Reply:
its a typo, removing comment